themes/edx.org/lms/templates/header.html

```diff
@@ -69,10 +69,9 @@ <h2 class="course-header">
 
     % if user.is_authenticated():
       % if not course or disable_courseware_header:
-        % if not nav_hidden or show_program_listing:
           <nav aria-label="Main" class="nav-main">
             <ul class="left list-inline nav-global authenticated">
-              % if not nav_hidden:
+              % if not nav_hidden and not show_program_listing:
```

Comment:
@tasawernawaz My concern is not on this line. My concern is on line #87.
According to line #87, if both "nav_hidden" and "show_program_listing" are true, the navigation header will show. Is that intended?